stream-chat-android-core/src/main/java/io/getstream/chat/android/client/call/CoroutineCall.kt

```diff
@@ -12,17 +12,23 @@ import kotlinx.coroutines.withContext
 @InternalStreamChatApi
 public class CoroutineCall<T : Any>(
     private val scope: CoroutineScope,
-    public val suspendingTask: suspend () -> Result<T>,
+    private val suspendingTask: suspend () -> Result<T>,
 ) : Call<T> {
 
     private var job: Job? = null
 
+    internal suspend fun awaitImpl(): Result<T> {
+        return withContext(scope.coroutineContext) {
+            suspendingTask()
+        }
```

Comment:
Applies the context from the `scope` before running `suspendingTask`. This way, it'll run on the specified dispatcher (among other things), giving us a chance to put the `suspendingTask` on a background thread even if it's not a well-behaving suspending function, like the ones we used to have in `QueryChannelsPostponeHelper`, marked as `suspend` but actually blocking.